app/models/miq_expression.rb

```diff
@@ -430,11 +430,11 @@ def self._to_human(exp, options = {})
 
   def to_ruby(tz = nil)
     tz ||= "UTC"
-    @ruby ||= self.class._to_ruby(@exp.deep_clone, @context_type, tz)
+    @ruby ||= _to_ruby(@exp.deep_clone, @context_type, tz)
     @ruby.dup
   end
 
-  def self._to_ruby(exp, context_type, tz)
+  def _to_ruby(exp, context_type, tz)
```

Comment:
This `exp` is now shadowing the existing attr_reader for `@exp`. While it may work fine, it might be better not to do this. Perhaps you could rename it as part of this PR? The longer term fix is for this method not to mutate this argument, so that we don't have to deep_clone it, and then the argument could disappear. But that may be way beyond the scope of this PR.